zkvm/src/constraints.rs

```diff
@@ -44,6 +45,42 @@ pub struct CommitmentWitness {
     blinding: Scalar,
 }
 
+impl Constraint {
+    pub fn verify<CS: r1cs::ConstraintSystem>(self, cs: &mut CS) -> Result<(), VMError> {
+        cs.specify_randomized_constraints(move |cs| {
+            let z = cs.challenge_scalar(b"verify challenge");
+
+            // Flatten the constraint into one expression
+            // Note: cloning because we can't move out of captured variable in an `Fn` closure
```

Comment:
And Box<FnOnce> is not supported yet. Cf. https://github.com/dalek-cryptography/bulletproofs/issues/244

(maybe add this reference to this comment)

Reply:
👍 Added comment!